lib/chat_web/live/main_live/index.html.heex

```diff
@@ -361,7 +361,7 @@
               <span class="text-xs">Admin</span>
             </button>
           <% end %>
-          <%= if assigns[:cargo_room][:status] == :complete do %>
+          <%= if assigns[:cargo_room][:status] == :complete and assigns[:cargo_room][:pub_key] in Map.keys(@room_map) do %>
```

Comment:
`in Map.keys()` is highly inefficient. You are making list from a map and traversing each key

`Map.has_key?` is better here